src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2237,103 +2254,115 @@ def _update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
-                                   'before enabling it again.')
-                addon_profile = ManagedClusterAddonProfile(
-                    enabled=True, config={})
-            elif addon == CONST_CONFCOM_ADDON_NAME:
-                if addon_profile.enabled:
-                    raise CLIError('The confcom addon is already enabled for this managed cluster.\n'
-                                   'To change confcom configuration, run '
-                                   f'"az aks disable-addons -a confcom -n {name} -g {resource_group_name}" '
-                                   'before enabling it again.')
-                addon_profile = ManagedClusterAddonProfile(
-                    enabled=True, config={CONST_ACC_SGX_QUOTE_HELPER_ENABLED: "false"})
-                if enable_sgxquotehelper:
-                    addon_profile.config[CONST_ACC_SGX_QUOTE_HELPER_ENABLED] = "true"
-            elif addon == CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME:
-                if addon_profile.enabled:
-                    raise CLIError('The azure-keyvault-secrets-provider addon is already enabled for this managed cluster.\n'
-                                   'To change azure-keyvault-secrets-provider configuration, run '
-                                   f'"az aks disable-addons -a azure-keyvault-secrets-provider -n {name} -g {resource_group_name}" '
-                                   'before enabling it again.')
-                addon_profile = ManagedClusterAddonProfile(
-                    enabled=True, config={CONST_SECRET_ROTATION_ENABLED: "false", CONST_ROTATION_POLL_INTERVAL: "2m"})
-                if enable_secret_rotation:
-                    addon_profile.config[CONST_SECRET_ROTATION_ENABLED] = "true"
-                if disable_secret_rotation:
-                    addon_profile.config[CONST_SECRET_ROTATION_ENABLED] = "false"
-                if rotation_poll_interval is not None:
-                    addon_profile.config[CONST_ROTATION_POLL_INTERVAL] = rotation_poll_interval
-                addon_profiles[CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME] = addon_profile
-            addon_profiles[addon] = addon_profile
+        if addon == CONST_WEB_APPLICATION_ROUTING_ADDON_NAME:
+            # web app routing settings are in ingress profile, not addon profile, so deal
+            # with it separately
+            if instance.ingress_profile is None:
+                instance.ingress_profile = ManagedClusterIngressProfile()
+            if instance.ingress_profile.web_app_routing is None:
+                instance.ingress_profile.web_app_routing = ManagedClusterIngressProfileWebAppRouting()
+            instance.ingress_profile.web_app_routing.enabled = enable
+
+            if dns_zone_resource_id is not None:
+                instance.ingress_profile.web_app_routing.dns_zone_resource_id = dns_zone_resource_id
```

Comment:
Similarly, we may check `addon_arg` directly.

Reply:
Updated code to check "addon_arg" directly without depending on the value.